operator/v1/types_ingress.go

```diff
@@ -803,9 +803,9 @@ type AWSClassicLoadBalancerParameters struct {
 
 // AWSNetworkLoadBalancerParameters holds configuration parameters for an
 // AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
-// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController,rule=`has(self.subnets) && has(self.subnets.ids) && has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids + self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
-// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController,rule=`has(self.subnets) && has(self.subnets.ids) && !has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
-// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController,rule=`has(self.subnets) && has(self.subnets.names) && !has(self.subnets.ids) && has(self.eipAllocations) ? size(self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=SetEIPForNLBIngressController;IngressControllerLBSubnetsAWS,rule=`has(self.subnets) && has(self.subnets.ids) && has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids + self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
```

Comment:
Have you tested this? I think we added a separate field, so it may need to be
```suggestion
// +openshift:validation:FeatureGateAwareXValidation:requiredFeatureGate=SetEIPForNLBIngressController;IngressControllerLBSubnetsAWS,rule=`has(self.subnets) && has(self.subnets.ids) && has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids + self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
```

Reply:
Yea sorry, I was trying to change to use `requiredFeatureGate`, but now I get:
```
F0103 11:03:44.606428 1073642 root.go:64] Error running codegen: 
error running generator schemapatch on group imageregistry.operator.openshift.io: 
	could not run schemapatch generator for group/version imageregistry.operator.openshift.io/v1: 
		could not execute schemapatch for manifest /home/gspence/src/github.com/openshift/api/imageregistry/v1/zz_generated.featuregated-crd-manifests/configs.imageregistry.operator.openshift.io/AAA_ungated.yaml: 
			github.com/openshift/api/imageregistry/v1:-: unable to locate schema for type "github.com/openshift/api/operator/v1".OperatorStatus
			github.com/openshift/api/imageregistry/v1:-: unable to locate schema for type "github.com/openshift/api/operator/v1".OperatorSpec
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:806:1: missing argument "featureGate" (at <input>:1:337)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:807:1: missing argument "featureGate" (at <input>:1:317)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:808:1: missing argument "featureGate" (at <input>:1:319)
			github.com/openshift/api/operator/v1:-: unknown type "github.com/openshift/api/operator/v1".OperatorSpec
			github.com/openshift/api/operator/v1:-: unknown type "github.com/openshift/api/operator/v1".OperatorStatus
error running generator schemapatch on group operator.openshift.io: 
	could not run schemapatch generator for group/version operator.openshift.io/v1alpha1: 
		could not execute schemapatch for manifest /home/gspence/src/github.com/openshift/api/operator/v1alpha1/zz_generated.featuregated-crd-manifests/clusterversionoperators.operator.openshift.io/ClusterVersionOperatorConfiguration.yaml: 
			github.com/openshift/api/operator/v1alpha1:-: unable to locate schema for type "github.com/openshift/api/operator/v1".LogLevel
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:806:1: missing argument "featureGate" (at <input>:1:337)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:807:1: missing argument "featureGate" (at <input>:1:317)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:808:1: missing argument "featureGate" (at <input>:1:319)
			github.com/openshift/api/operator/v1:-: unknown type "github.com/openshift/api/operator/v1".LogLevel
	could not run schemapatch generator for group/version operator.openshift.io/v1: 
		could not execute schemapatch for manifest /home/gspence/src/github.com/openshift/api/operator/v1/manual-override-crd-manifests/clustercsidrivers.operator.openshift.io/AAA_ungated.yaml: 
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:806:1: missing argument "featureGate" (at <input>:1:337)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:807:1: missing argument "featureGate" (at <input>:1:317)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:808:1: missing argument "featureGate" (at <input>:1:319)
error running generator schemapatch on group samples.operator.openshift.io: 
	could not run schemapatch generator for group/version samples.operator.openshift.io/v1: 
		could not execute schemapatch for manifest /home/gspence/src/github.com/openshift/api/samples/v1/zz_generated.featuregated-crd-manifests/configs.samples.operator.openshift.io/AAA_ungated.yaml: 
			github.com/openshift/api/samples/v1:-: unable to locate schema for type "github.com/openshift/api/operator/v1".ManagementState
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:806:1: missing argument "featureGate" (at <input>:1:337)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:807:1: missing argument "featureGate" (at <input>:1:317)
			/home/gspence/src/github.com/openshift/api/operator/v1/types_ingress.go:808:1: missing argument "featureGate" (at <input>:1:319)
			github.com/openshift/api/operator/v1:-: unknown type "github.com/openshift/api/operator/v1".ManagementState
```

I'm not totally sure why I'm getting `missing argument "featureGate"`, I might be doing something silly...any help is appreciated.